public/app/features/scenes/components/SceneFlexLayout.tsx

```diff
@@ -2,19 +2,29 @@ import React, { CSSProperties } from 'react';
 
 import { Field, RadioButtonGroup } from '@grafana/ui';
 
-import { SceneObjectBase } from '../core/SceneObjectBase';
+import { Defaultize, SceneObjectBase } from '../core/SceneObjectBase';
 import { SceneObjectSize, SceneLayoutState, SceneComponentProps, SceneLayoutChild } from '../core/types';
 
 export type FlexLayoutDirection = 'column' | 'row';
-
 interface SceneFlexLayoutState extends SceneLayoutState {
-  direction?: FlexLayoutDirection;
+  direction: FlexLayoutDirection;
 }
 
+const defaultProps = {
+  direction: 'row' as const,
+};
+
 export class SceneFlexLayout extends SceneObjectBase<SceneFlexLayoutState> {
   static Component = FlexLayoutRenderer;
   static Editor = FlexLayoutEditor;
 
+  constructor(state: Defaultize<SceneFlexLayoutState, typeof defaultProps>) {
+    super({
+      ...defaultProps,
+      ...state,
+    });
+  }
+
```

Comment:
this bit works, but would be nice if this was handled / infered by base class typing